docs/migrate-from-legacy-sdks-to-app-bridge.md

```diff
@@ -0,0 +1,248 @@
+
+#### Pagination
+
+Adding pagination to the title bar using `ShopifyApp.Bar.initialize` and `ShopifyApp.Bar.setPagination` has been deprecated. We recommend implenting pagination within your app’s UI. If you’re using Polaris, you can use the pagination controls provided by the Polaris Page component.
+
+If you prefer to keep pagination in your app’s titlebar, you can use Shopify App Bridge to add pagination buttons to the titlebar.
+
+##### Use Shopify App Bridge to add pagination buttons to the title bar
+
+[Learn more about the TitleBar action set](https://shopify.dev/apps/tools/app-bridge/actions/titlebar).
+
+```js
+import createApp from '@shopify/app-bridge';
+import { TitleBar, Button } from '@shopify/app-bridge/actions';
+
+const app = createApp({
+  apiKey: '12345',
+  host: host,
+});
+
+const paginationPreviousButton = Button.create(app, { label: '←' })
+paginationPreviousButton.subscribe(Button.Action.CLICK, function() {
+  // pagination previous action
+})
+
+const paginationNextButton = Button.create(app, { label: '→' })
+paginationNextButton.subscribe(Button.Action.CLICK, function() {
+  // pagination next action
+})
+
+const titleBarOptions = {
+  buttons: {
+    secondary: [paginationPreviousButton, paginationNextButton],
+  }
+};
+const myAppTitleBar = TitleBar.create(app, titleBarOptions);
+```
+
+##### Use the Polaris Page component pagination controls
+
+[Learn more about the Polaris Page component](https://polaris.shopify.com/components/structure/page/pagination).
+
+```jsx
+import React, {useCallback} from 'react';
+import {Page} from '@shopify/polaris';
+
+function myApp() {
+
+  const paginationPrevious = useCallback(() => {
+    // pagination previous action
+  }, []);
+
+  const paginationNext = useCallback(() => {
+    // pagination next action
+  }, []);
+
+  const paginationOptions = {
+    hasNext: true,
+    hasPrevious: true,
+    onNext: paginationNext,
+    onPrevious: paginationPrevious
+  };
+
+  return (
+    <Page title="My App" pagination={paginationOptions}>
+      App
+    </Page>
+  );
+}
+```
+
+
+#### protocol URL parameter
+
+Since all embedded apps must use HTTPS, the `protocol` parameter is no longer relevant.
+
+### Keeping the frame URL in sync
+
+The EASDK `initialize` function automatically updates the Shopify admin URL.
+
+Since Shopify App Bridge is designed to accomodate a broader range of uses, it can’t make this assumption. It’s only aware of navigation using [History](https://shopify.dev/apps/tools/app-bridge/actions/navigation/history) and [Redirect](https://shopify.dev/apps/tools/app-bridge/actions/navigation/redirect) actions.
+
+If your app uses links or server-side redirects for navigation, you can keep the URL in sync by dispatching a `History.Action.PUSH` action wherever you called EASDK’s `initialize` method (for example, after `createApp()`).
+
```

Comment:
If your app is a React single-page app, you can use the [Route Propagator component ](https://shopify.dev/apps/tools/app-bridge/react-components/route-propagator)

Reply:
I'll add that.